pkg/pquerier/collector.go

```diff
@@ -263,3 +273,67 @@ func aggregateClientAggregatesCrossSeries(ctx *selectQueryContext, res *qryResul
 	lastT, lastV := it.At()
 	return lastT, lastV, nil
 }
+
+func intervalAggregation(ctx *selectQueryContext, res *qryResults, t int64, v float64) {
+	currentCell := (t - ctx.queryParams.From) / res.query.aggregationParams.Interval
```

Comment:
do not repeat - move to a function

Reply:
that would be a one liner with 3 parameters.. not sure if it would simplify the code

Reply:
it will not simplify but will localise the possible changes in the logic and stop duplicating same code